app/src/main/java/com/kodeco/android/coordplot/MainActivity.kt

```diff
@@ -0,0 +1,94 @@
+package com.kodeco.android.coordplot
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.Slider
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableFloatStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.kodeco.android.coordplot.ui.theme.MyApplicationTheme
+
+class MainActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            MyApplicationTheme {
+                PlotSurface()
+            }
+        }
+    }
+}
+
+@Composable
+fun PlotSurface() {
+    Surface(
+        modifier = Modifier.fillMaxSize(),
+        color = Color.White
+    ) {
+        Column (modifier = Modifier.fillMaxSize(),
+                horizontalAlignment = Alignment.CenterHorizontally)
+            {
+            var xPercentage: Float by remember { mutableFloatStateOf(0.5f) }
+            var yPercentage: Float by remember { mutableFloatStateOf(0.5f) }
+            val boxSize = 300
+            val innerBoxSize = 20
+            val xOffset = 10
+            val yOffset = 10
+
+            Map(xPercentage, yPercentage, modifier = Modifier.size(boxSize.dp)
+                .background(Color.LightGray)
+                .offset((xPercentage * boxSize - xOffset).dp, (yPercentage * boxSize - yOffset).dp),
+                innerBoxSize)
```

Comment:
Nice work here on exposing the params here in the `Map` composable function!

One thing to note - it may be worth considering placing the logic for setting the modifier directly _inside_ the Map itself, rather than exposing a `Modifier` property itself.  This helps with things like encapsulation and reusability. There's a lot a functionality that a `Modifier` can do.  In this case, `Map()` will always need to apply the `offset()` modifier in order for the inner circle to move around.  If the call site here did not add the `offset()` modifier, then it wouldn't happen.

Reply:
Great work on chaining the modifier parameters.
Would agree with Zac, and it would be even more secular if the Map rendering is done within Map, so that it can be re-used anywhere